### PR TITLE
AP_DDS: add publisher for rosgraph_msgs/msg/Clock

### PR DESCRIPTION
**Pull request by srmainwaring**
Add a publisher for `rosgraph_msgs/msg/Clock` on topic `/ap/clock`.

ROS 2 nodes use the [`rosgraph_msgs/msg/Clock`](http://docs.ros.org/en/melodic/api/rosgraph_msgs/html/msg/Clock.html) message published on `/clock` to control the behaviour of time accessed via `Node.get_clock).now()` depending on the value of the parameter `use_sim_time`. 

The built-in type `builtin_interfaces/msg/Time` is not usually exposed as a ROS topic, but generally contained inside another message type - either `Clock` or in `Header`. Rather than remove publishing the `Time` message, it has been mapped instead to the topic `ap/time`.

## Details

- Publish `builtin_interfaces/msg/Time` to `/ap/time` (renamed from `/ap/clock`).
- Publish `rosgraph_msgs/msg/Clock` to `/ap/clock`.

## Testing

Launch SITL with the micro ROS agent:

```bash
ros2 launch ardupilot_sitl sitl_dds.launch.py tty0:=./dev/ttyROS0 tty1:=./dev/ttyROS1 refs:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/dds_xrce_profile.xml baudrate:=115200 device:=./dev/ttyROS0 synthetic_clock:=True wipe:=False model:=quad speedup:=1 slave:=0 instance:=0 uartC:=uart:./dev/ttyROS1 defaults:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/copter.parm,$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/dds.parm sim_address:=127.0.0.1 master:=tcp:127.0.0.1:5760 sitl:=127.0.0.1:5501
```

You may need to  `param set DDS_ENABLE 1` using MAVPRoxy then relaunch. 

Verify topics:

```bash
ros2 topic list -v
Published topics:
 * /ap/battery/battery0 [sensor_msgs/msg/BatteryState] 1 publisher
 * /ap/clock [rosgraph_msgs/msg/Clock] 1 publisher
 * /ap/geopose/filtered [geographic_msgs/msg/GeoPoseStamped] 1 publisher
 * /ap/navsat/navsat0 [sensor_msgs/msg/NavSatFix] 1 publisher
 * /ap/pose/filtered [geometry_msgs/msg/PoseStamped] 1 publisher
 * /ap/tf_static [tf2_msgs/msg/TFMessage] 1 publisher
 * /ap/time [builtin_interfaces/msg/Time] 1 publisher
 * /ap/twist/filtered [geometry_msgs/msg/TwistStamped] 1 publisher
 * /parameter_events [rcl_interfaces/msg/ParameterEvent] 1 publisher
 * /rosout [rcl_interfaces/msg/Log] 1 publisher

Subscribed topics:
```

Echo topic:

```bash
ros2 topic echo /ap/clock
clock:
  sec: 5
  nanosec: 328701000
---
```

## References

- https://design.ros2.org/articles/clock_and_time.html
- http://wiki.ros.org/Clock
- http://docs.ros.org/en/melodic/api/rosgraph_msgs/html/msg/Clock.html

### CODE REVIEW

File: libraries/AP_DDS/README.md [hunk 1]
```diff
@@ -152,23 +152,27 @@ After your setups are complete, do the following:
   /Ardupilot_DDS_XRCE_Client
 
   $ ros2 topic list  -v
+  ros2 topic list -v
```
Comment (Ryanf55) [inline libraries/AP_DDS/README.md:155]: Extra ros2 topic list here.

**Review by Ryanf55 — CHANGES_REQUESTED** [review]
LGTM, tested and it works, please just update the small README duplicate line. Great work! 

```
---
clock:
  sec: 1683862472
  nanosec: 423896000
---

```

File: libraries/AP_DDS/README.md (hunk 1 above)
Comment (srmainwaring) [reply libraries/AP_DDS/README.md]: Updated in cd15131eab

**Comment by srmainwaring** [comment]
Thanks for the review @Ryanf55.

Also updated the time msg test in f1a6afbb02 as I'd missed updating the function name and doc string for `test_dds_serial_time_msg_recv` and `test_dds_udp_time_msg_recv`.

**Review by tridge — APPROVED** [review]

**Pull request merged**